src/firewall/core/fw_zone.py

```diff
@@ -95,9 +95,15 @@ def get_zone_of_source(self, source):
                 return zone
         return None
 
-    def get_zone(self, zone):
-        z = self._fw.check_zone(zone)
-        return self._zones[z]
+    def check_zone(self, zone):
+        return self.get_zone(zone).name
+
+    def get_zone(self, zone, required=True):
+        zname = self._fw.get_default_zone(zone)
+        z = self._zones.get(zname, None)
+        if z is None and required:
+            raise FirewallError(errors.INVALID_ZONE, zone)
+        return z
```

Comment:
`get_default_zone()` is no longer changed.

The only new thing is that `fw_zone.py` now also calls `self._fw.get_default_zone()`.

However, `fw_zone.py` already has many uses of `self._fw`. So that is common.